db/migrate/20170706201604_add_source_id_to_user.rb

```diff
@@ -0,0 +1,10 @@
+class AddSourceIdToUser < ActiveRecord::Migration
+  def change
+    add_reference :users, :source, index: true, foreign_key: true
+    User.find_each do |u|
+      s = Source.where(user_id: u.id).first
+      u.source_id = s.id
+      u.save!
```

Comment:
I prefer to use `update_columns` in migrations instead of `save!`, because we usually don't want the callbacks to be called. Also, `update_columns` is faster. But if you have a reason to use `save!`, let me know plz.